packages/esm-service-queues-app/src/queue-services/queue-service-form.workspace.tsx

```diff
@@ -138,31 +153,37 @@ const QueueServiceForm: React.FC<DefaultWorkspaceProps> = ({ closeWorkspace }) =
 
         <Column>
           <Layer className={styles.input}>
-            <Select
-              id="location"
-              invalidText="Required"
-              labelText={t('selectALocation', 'Select a location')}
-              onChange={(event) => setUserLocation(event.target.value)}
-              value={userLocation}>
-              {!userLocation && <SelectItem text={t('selectALocation', 'Select a location')} value="" />}
-              {queueLocations.length === 0 && (
-                <SelectItem text={t('noLocationsAvailable', 'No locations available')} value="" />
+            <Controller
+              name="userLocation"
+              control={control}
+              render={({ field }) => (
+                <Select
+                  {...field}
+                  id="location"
+                  invalidText="Required"
+                  labelText={t('selectALocation', 'Select a location')}>
+                  {!field.value && <SelectItem text={t('selectALocation', 'Select a location')} value="" />}
+                  {queueLocations.length === 0 && (
+                    <SelectItem text={t('noLocationsAvailable', 'No locations available')} value="" />
+                  )}
```

Comment:
```suggestion
```

I don't think we should do this. Maybe it makes more sense to disable the select input and show a notification below if there are no queue locations.